src/parse-output-path-and-format.ts

```diff
@@ -0,0 +1,27 @@
+const mime = require("mime-types");
+const path = require("path");
+
+export function parseOutputPathAndFormat(output: string, format: string) {
+  const extension = path.extname(output);
+  const extensionMimeType = mime.lookup(extension);
+
+  if (extensionMimeType === format) {
+    return [output, format];
+  }
+
+  const formatExtension = mime.extension(format);
```

Comment:
I wonder if it is worth also adding the ability to infer format from extension. i.e. `.jpg` and we know it needs to be `image/jpeg`

Reply:
Yeah this happens now. Although it's not clear from the code. Let me actually make it a bit cleaner...

Currently we have this:
```typescript
  if (!extensionMimeType) {
    const extension = formatExtension === "jpeg" ? "jpg" : formatExtension;

    return [`${output}.${extension}`, format];
  }

  return [output, extensionMimeType];
```

I'll remove the negation and check that the extension produces a valid format.